src/Chunk.cpp

```diff
@@ -194,10 +194,28 @@ void cChunk::MarkRegenerating(void)
 
 
 
+bool cChunk::HasPlayerEntities()
+{
+	bool hasPlayerEntities {false};
+	for (auto & Entity: m_Entities)
+	{
+		if (Entity->IsPlayer())
+		{
+			hasPlayerEntities = true;
+		}
+	}
+	return hasPlayerEntities;
+}
+
+
+
+
+
 bool cChunk::CanUnload(void)
 {
 	return
 		m_LoadedByClient.empty() &&  // The chunk is not used by any client
+		!HasPlayerEntities() &&      // Ensure not only the absence of ClientHandlers, but also of cPlayer objects
```

Comment:
This should be added to `CanUnloadAfterSaving` as well.

Reply:
Youre right!